src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,63 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { User } from '../../types/User';
+import { useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { useDispatch } from 'react-redux';
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useDispatch();
+
+      <Loader />
+  useEffect(() => {
+    getUser(currentTodo?.userId)
+      .then(setUser)
+      .finally(() => setLoading(false));
+  }, [currentTodo]);
+  const closingTodo = () => {
+    dispatch(currentTodoSlice.actions.closeTodo(''));
+  };
```

Comment:
```suggestion
  useEffect(() => {
    getUser(currentTodo?.userId)
      .then(setUser)
      .finally(() => setLoading(false));
  }, [currentTodo]);

  const closingTodo = () => {
    dispatch(currentTodoSlice.actions.closeTodo(''));
  };

```
Loader is redundant here)
Also, add blank lines